cmd/imports/ethereum.go

```diff
@@ -0,0 +1,75 @@
+package imports
+
+import (
+	"context"
+
+	"github.com/consensys/quorum-key-manager/pkg/errors"
+	"github.com/consensys/quorum-key-manager/pkg/json"
+	"github.com/consensys/quorum-key-manager/src/infra/log"
+	manifest "github.com/consensys/quorum-key-manager/src/infra/manifests/entities"
+	"github.com/consensys/quorum-key-manager/src/stores"
+	"github.com/consensys/quorum-key-manager/src/stores/database"
+	"github.com/consensys/quorum-key-manager/src/stores/database/models"
+	"github.com/consensys/quorum-key-manager/src/stores/entities"
+	eth "github.com/consensys/quorum-key-manager/src/stores/manager/ethereum"
+)
+
+func ImportEthereum(ctx context.Context, db database.ETHAccounts, mnf *manifest.Manifest, logger log.Logger) error {
+	logger.Info("importing ethereum accounts...", "store", mnf.Kind, "store_name", mnf.Name)
+
+	store, err := getEthStore(mnf, logger)
+	if err != nil {
+		return err
+	}
+
+	storeIDs, err := store.List(ctx, 0, 0)
+	if err != nil {
+		return err
+	}
+
+	dbAddresses, err := db.SearchAddresses(ctx, false, 0, 0)
+	if err != nil {
+		return err
+	}
+
+	var n uint
+	for _, id := range storeIDs {
+		key, err := store.Get(ctx, id)
+		if err != nil {
+			return err
```

Comment:
Maybe we should continue here and mark failures

Reply:
discussed